src/components/IncorporationAgreement/AgreementType.vue

```diff
@@ -0,0 +1,150 @@
+
+        <!-- Summary Warning -->
+        <div v-if="showErrorSummary" class="agreement-invalid-message">
+            <span>
+            <v-icon color="#1976d2">mdi-information-outline</v-icon>
+            This step is not complete.
+            <router-link id="router-link" :to="{ path: '/incorporation-agreement' }">
+                Return to this step to complete it.
+            </router-link>
+            </span>
+        </div>
+        <!-- Summary Content -->
+        <div v-else class="summary-desc">
+            <div><v-icon color="green" class="agreement-valid-icon">mdi-check</v-icon></div>
+             <div>
+                {{ selectedAgreementDescription }}
+            </div>
+        </div>
+      </v-card>
+    </div>
+    <div v-else style="margin-left:1rem;">
+      <v-card flat>
+        <v-radio-group v-model="agreementType" @change="changeAgreementType" class="agreement-option-list">
+          <v-radio v-for="(item, index) in incorporationAgreementTypes"
+           :key="index" :value="item.code" :id="`agreement-type-${item.code}`">
+            <template slot="label">
+              <div v-html="item.description" class="agreement-option"/>
+            </template>
+          </v-radio>
+        </v-radio-group>
+      </v-card>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Component, Vue, Prop } from 'vue-property-decorator'
+import { Action, State } from 'vuex-class'
+
+// Interfaces
+import { ActionBindingIF } from '@/interfaces'
+import { AgreementTypeResource } from '@/resources'
+
+@Component
+export default class AgreementType extends Vue {
+   private incorporationAgreementTypes = AgreementTypeResource
+
+  // State
+  @State(state => state.stateModel.incorporationAgreementStep.agreementType)
+  readonly agreementTypeState: string | null
+
+  @Prop({ default: false })
+  private showErrorSummary: boolean
+
+  @Prop({ default: false })
+  private isSummary: boolean
+
+  // Actions
+  @Action setIncorporationAgreementStepData!: ActionBindingIF
+  @Action setIgnoreChanges!: ActionBindingIF
+
+  private agreementType: string | null = null
+
+  // Lifecycle methods
+  private created (): void {
+    // temporarily ignore data changes
+    this.setIgnoreChanges(true)
+    this.agreementType = this.agreementTypeState
+    // watch data changes once page has loaded (in next tick)
+    Vue.nextTick(() => {
+      this.setIgnoreChanges(false)
+    })
+  }
+
+  mounted (): void {
+    this.setIncorporationAgreementStepData({
+      valid: !!this.agreementType,
+      agreementType: this.agreementType
+    })
+  }
+
+  private changeAgreementType (): void {
+    this.setIncorporationAgreementStepData({
+      valid: !!this.agreementType,
+      agreementType: this.agreementType
+    })
+  }
+
+  private get selectedAgreementDescription () : string {
+    if (this.agreementTypeState) {
+      return this.incorporationAgreementTypes.find(item => item.code === this.agreementTypeState)
```

Comment:
If the code is not found, this will return 'undefined'. Would it be better to return '' ?

Reply:
I dont think the above situation will occur at all

Reply:
You're right, it should never happen. But I always encourage safe code (esp since JS just exits a method when it encounters an error so the user might not know the page is incomplete).

Not a deal-breaker. I will accept this PR without this.